src/ui/ui-storyEditor.js

```diff
@@ -68,14 +68,7 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
             onVideoBlockAdditionRequested: null,
             onRemoveBlocksRequested: null,
             onRemoveBlocksCompleted: null,
-            onEditorTemplateRendered: null,
-            onBlockManagerCreated: null,
-            onReadyToBind: {
-                events: {
-                    onEditorTemplateRendered: "{that}.events.onEditorTemplateRendered",
-                    onBlockManagerCreated: "{that}.events.onBlockManagerCreated"
-                }
-            }
+            onBlockOrderUpdated: null
```

Comment:
I began to work on this and added some explicit test cases that check the story contents before and after each reorder, but in doing so I surfaced a previously-unknown bug: it seems that there is something happening with the changeApplier call in `sjrk.storyTelling.ui.storyEditor.sortStoryContent` that is merging the contents of that value rather than replacing it. The result is that blocks in the `story`'s content array have nearly every possible key present, i.e. text blocks have their usual fields in addition to `imageUrl` **and** `mediaUrl`, `fileDetails`, `alternativeText`, etc., and this ends up in the database on publish.

I have tried, among other things:
- calling the changeApplier with "DELETE" on the `content` key before setting the new value
- bundling the individual block component order value updates as a single change transaction
- using `fluid.set` to set the `content` model value explicitly
- specifying the story grade's `mergePolicy` as `nomerge` (maybe this should have been `replace`...)

The consequences are minimal, basically only that the test story will look strange since certain block types will have unexpected fields, though it is quite strange.

Reply:
The fact that it is "accumulating" is a well-known design feature of the ChangeApplier since its earliest days - whether it is the best choice or not is secondary to the issue that doing anything about it at this point would break thousands of lines of code depending on it.
The correct approach is a combination of your first and second points - as seen in utilities such as this one in the prefs framework store - https://github.com/fluid-project/infusion/blob/master/src/framework/preferences/js/Store.js#L245 - this is such a frequently used pattern that it should be a core framework utility.
Point 3, using fluid.set directly into the model must never be operated since it will corrupt the ChangeApplier's records, and point 4 is inadvisable since mergePolicies are generally framework internals that should only be used as a last resort - their operation is fragile and not terribly well documented.
I agree we should get rid of this event.

Reply:
The solution turned out to be to add a `DELETE` operation before updating the story contents in `sjrk.storyTelling.ui.storyUi.updateStoryFromBlocks`, not in `sjrk.storyTelling.ui.storyEditor.sortStoryContent`. I have wrapped up all changes in transactions either way, in order to economise model changes and avoid any unnecessary firing of the block-to-story model listener.